src/NumericalAlgorithms/Spectral/Mesh.hpp

```diff
@@ -254,6 +254,15 @@ class Mesh {
   std::array<Spectral::Quadrature, Dim> quadratures_{};
 };
 
+/*!
+ * \ingroup DataStructuresGroup
+ * \brief Returns `true` if the mesh is isotropic, `false` otherwise.
+ *
+ * If `Dim` is zero, then `true` is always returned.
+ */
+template <size_t Dim>
+bool is_isotropic(const Mesh<Dim>& mesh);
```

Comment:
Why not make this a member function of `Mesh` so you can do `mesh.is_isotropic()`? Might be more efficient because you can calculate it in the constructor and just store a bool. Don't know if that actually affects performance though.